content/blogs/2024-04-25-kestra-vs-airflow-orchestrating-python-scripts.md

```diff
@@ -0,0 +1,152 @@
+---
+title: "Kestra vs Airflow for orchestrating Python scripts"
+description: "This blog compares Kestra and Airflow for the fundamental task of orchestrating Python scripts."
+date: 2024-04-25T10:00:00
+category: Solutions
+author:
+  name: Shruti Mantri
+  image: "smantri"
+---
+
+This blog will compare Kestra vs Airflow for one of the most simple but fundamental tasks of orchestrating Python scripts.
+
+Let us begin with how the flow or job looks like in these orchestrators for running a simple Hello World Python script. In Airflow, the flow would look as follows:
+
+```python
+from __future__ import annotations
+
+import pendulum
+
+from airflow.models.dag import DAG
+from airflow.operators.python import PythonOperator
+
+with DAG(
+    dag_id="hello-world",
+    schedule=None,
+    start_date=pendulum.datetime(2021, 1, 1, tz="UTC"),
+    catchup=False,
+    tags=["example"],
+):
+    def hello_world(**kwargs):
+        print("Hello World")
+
+    python_task = PythonOperator(task_id="print_hello_world", python_callable=hello_world)
+```
+
+The similar workflow in Kestra would look as follows:
+
+```yaml
+id: hello-world
+namespace: dev
+description: Print Hello World
+
+tasks:
+  - id: python-script
+    type: io.kestra.plugin.scripts.python.Script
+    script: |
+      print("Hello World")
+```
+
+Not a major difference, isn't it?! But as the complexities creep in, you would start noticing some major differences.
```

Comment:
I'll cut that one - to me there is an important difference between the two (Airflow is way more verbose)